src/main/java/app/controller/AttachmentController.java

```diff
@@ -14,38 +19,64 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Attachment management", description = "Endpoints for managing attachments")
 @RestController
 @RequestMapping("/api/attachments")
 @RequiredArgsConstructor
 public class AttachmentController {
     private final AttachmentService attachmentService;
 
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
     @ResponseStatus(HttpStatus.CREATED)
     @PostMapping(consumes = "multipart/form-data")
+    @Operation(summary = "Create a new attachment",
+            description = "Create a new attachment for a task with task ID")
     public AttachmentResponseDto createAttachment(@RequestParam("filePath") String filePath,
-                                                  @RequestParam("taskId") Long taskId) {
-        return attachmentService.create(filePath, taskId);
+                                                  @RequestParam("taskId") Long taskId,
+                                                  Authentication authentication) {
+        return attachmentService.create(filePath, taskId, getUserId(authentication));
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
     @GetMapping("/{id}")
-    public AttachmentResponseDto getAttachmentById(@PathVariable Long id) {
-        return attachmentService.getById(id);
+    @Operation(summary = "Get attachment by ID",
+            description = "Get all information about the attachment by ID")
+    public AttachmentResponseDto getAttachmentById(@PathVariable Long id,
+                                                   Authentication authentication) {
+        return attachmentService.getById(id, getUserId(authentication));
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
     @GetMapping("/{id}/all")
-    public List<AttachmentResponseDto> getAllByTaskId(@PathVariable Long id) {
-        return attachmentService.getAllByTaskId(id);
+    @Operation(summary = "Get all attachments",
+            description = "Get all attachments by task ID")
+    public List<AttachmentResponseDto> getAllByTaskId(@PathVariable Long id,
+                                                      Authentication authentication) {
+        return attachmentService.getAllByTaskId(id, getUserId(authentication));
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")

```

Comment:
As there is only 1 user with ADMIN permissions and it will be added through the script, change te annotations to 
```suggestion
    @PreAuthorize("hasRole('ROLE_USER')")

```
because ADNMIN user will have both permissions